hopp/tools/design/wind/turbine_library_interface_tools.py

```diff
@@ -0,0 +1,200 @@
+            elif wind_plant.site.hub_height is not None:
+                if any(float(k) == float(wind_plant.site.hub_height) for k in turbine_specs["hub_height"]):
+                    hub_height = wind_plant.site.hub_height
+                    logger.info(f"multiple hub height options available for {turbine_name} turbine. Setting hub height to SiteInfo hub_height: {hub_height}")
+                    return hub_height
+            
+            # check the hub-height of PySAM wind turbine object
+            elif isinstance(wind_plant._system_model,windpower.Windpower):
+                if any(float(k) == float(wind_plant._system_model.Turbine.wind_turbine_hub_ht) for k in turbine_specs["hub_height"]):
+                    hub_height = wind_plant._system_model.Turbine.wind_turbine_hub_ht
+                    logger.info(f"multiple hub height options available for {turbine_name} turbine. Setting hub height to WindPower.WindPower.Turbine.wind_turbine_hub_ht: {hub_height}")
+                    return hub_height
+            
+            # set hub height as median from options
+            else:
+                hub_height = np.median(turbine_specs["hub_height"])
+                logger.info(f"multiple hub height options available for {turbine_name} turbine. Setting hub height to median available height: {hub_height}")
+                return hub_height
+                
+    else:
+        hub_height = turbine_specs["hub_height"]
+        if wind_plant.config.hub_height is not None:
+            if hub_height != wind_plant.config.hub_height:
+                logger.warning(f"turbine hub height ({hub_height}) does not equal wind_plant.config.hub_height ({wind_plant.config.hub_height})")
+        if hub_height != wind_plant.site.hub_height:
+            logger.warning(f"turbine hub height ({hub_height}) does not equal site_info.hub_height ({wind_plant.site.hub_height})")
+
+    return hub_height
+
+
+def get_pysam_turbine_specs(turbine_name,wind_plant):
+    """Load turbine data from turbine-models library to use with PySAM wind simulation.
+
+    Args:
+        turbine_name (str): name of turbine in turbine-models library
+        wind_plant (:obj:`hopp.simulation.technologies.wind.wind_plant.WindPlant`): wind plant object.
+
+    Raises:
+        ValueError: if turbine is missing data.
+
+    Returns:
+        dict: turbine model dictionary formatted for PySAM.
+    """
+    t_lib = Turbines()
+    turbine_specs = t_lib.specs(turbine_name)
+    if isinstance(turbine_specs,dict):
+        turbine_dict = extract_power_curve(turbine_specs, model_name = "pysam")
+
+        hub_height = check_hub_height(turbine_specs,wind_plant)
+        
+        turbine_dict.update({
+            "wind_turbine_rotor_diameter":turbine_specs["rotor_diameter"],
+            "wind_turbine_hub_ht":hub_height,
+            })
+
+    else:
+        raise ValueError(f"turbine {turbine_name} is missing some data, please try another turbines")
+    return turbine_dict
```

Comment:
The return line should replace the else statement and the last line of the function should be the error raising.